rts/motoko-rts/src/gc/generational/sanity_checks.rs

```diff
@@ -52,25 +52,27 @@ unsafe fn verify_heap(limits: &Limits) {
     assert!(SNAPSHOT.len().as_usize() <= limits.free);
     let mut pointer = limits.base;
     while pointer < SNAPSHOT.len().as_usize() {
-        let current = pointer as *mut Obj;
-        let previous = (SNAPSHOT.payload_addr() as usize + pointer) as *mut Obj;
-        assert!(current.tag() == previous.tag());
-        visit_pointer_fields(
-            &mut (),
-            current,
-            current.tag(),
-            0,
-            |_, current_field| {
-                if relevant_field(current_field, limits.last_free) {
-                    verify_field(current_field);
-                }
-            },
-            |_, slice_start, arr| {
-                assert!(slice_start == 0);
-                arr.len()
-            },
-        );
-        pointer += object_size(current as usize).to_bytes().as_usize();
+        if Value::from_ptr(pointer).is_obj() {
```

Comment:
There is a check in the `incremental/sanity_check.rs` in the higher PR (because I thought to only enable forwarding with that GC mode) that checks that no forwarding is used when the GC is not active (completed updating). There are additional checks in `types.rs` with `check_forwarding_pointer()` that the forwarding is maximum one indirection.